src/core/dense_set.cc

```diff
@@ -263,6 +304,41 @@ void DenseSet::Reserve(size_t sz) {
   }
 }
 
+void DenseSet::Fill(DenseSet* other) const {
+  DCHECK(other->entries_.empty());
+
+  other->Reserve(UpperBoundSize());
+
+  constexpr unsigned kArrLen = 32;
+  CloneItem arr[kArrLen];
+  unsigned len = 0;
+
+  for (auto it = entries_.begin(); it != entries_.end(); ++it) {
+    const DensePtr* ptr = &(*it);
+
+    if (!ptr->IsEmpty()) {
+      arr[len].has_ttl = ptr->HasTtl();
+
+      if (ptr->IsObject()) {
+        arr[len].link = nullptr;
+        arr[len].obj = ptr->Raw();
+        __builtin_prefetch(arr[len].obj, 0, 1);
```

Comment:
I've never used it before, have you measured what is the impact?

Reply:
I'd add a brief comment why the function looks the way it does, i.e. we parallelize data dependency